core/src/test/java/org/hibernate/ogm/sessionfactory/SessionFactoryInlineTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.sessionfactory;
+
+import static junit.framework.TestCase.fail;
+
+import java.util.function.Consumer;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+/**
+ * Helps to test {@link SessionFactory} creation
+ * providing an inline test API
+ *
+ * @author Fabio Massimo Ercoli
+ */
+public class SessionFactoryInlineTest {
+
```

Comment:
I don't like the name of this class, it's misleading because it's not a Test.
Also, I don't find the methods in it all that useful.

The problem for me is that some methods simply replace existing approaches without adding much, meaning that becomes something more to learn for new users. Feel free to challenge this

What I would do is to rename this class to something like `JPATestHelper` or `HibernateTestHelper` and keep only the `inTransaction` method . We could also move the existing method in `OgmJpaTestCase` and `OgmTestCase` here so that we have everything in one place and we can use it even when the test class doesn't extends those classes.

Let me know what you think